apps/api/src/modules/archived-notifications/archived-notifications.service.ts

```diff
@@ -0,0 +1,103 @@
+import { forwardRef, Inject, Injectable, Logger } from '@nestjs/common';
+import { ArchivedNotification } from './entities/archived-notification.entity';
+import { DataSource, QueryRunner } from 'typeorm';
+import { Notification } from 'src/modules/notifications/entities/notification.entity';
+import { NotificationsService } from '../notifications/notifications.service';
+import { ConfigService } from '@nestjs/config';
+
+@Injectable()
+export class ArchivedNotificationsService {
+  protected readonly logger = new Logger(ArchivedNotificationsService.name);
+
+  constructor(
+    @Inject(forwardRef(() => NotificationsService))
+    protected readonly notificationsService: NotificationsService,
+    private readonly configService: ConfigService,
+    private dataSource: DataSource,
+  ) {}
+
+  private convertToArchivedNotifications(notifications: Notification[]): ArchivedNotification[] {
+    return notifications.map((notification) => {
+      const archivedNotification = new ArchivedNotification();
+      archivedNotification.applicationId = notification.applicationId;
+      archivedNotification.channelType = notification.channelType;
+      archivedNotification.createdBy = notification.createdBy;
+      archivedNotification.createdOn = notification.createdOn;
+      archivedNotification.data = notification.data;
+      archivedNotification.deliveryStatus = notification.deliveryStatus;
+      archivedNotification.notificationId = notification.id;
+      archivedNotification.providerId = notification.providerId;
+      archivedNotification.result = notification.result;
+      archivedNotification.retryCount = notification.retryCount;
+      archivedNotification.updatedBy = notification.updatedBy;
+      archivedNotification.updatedOn = notification.updatedOn;
+      archivedNotification.status = notification.status;
+
+      this.logger.debug(
+        `Preparing ArchivedNotification entry using NotificationID: ${archivedNotification.notificationId}, DeliveryStatus: ${archivedNotification.deliveryStatus}, ApplicationID: ${archivedNotification.applicationId}, ProviderID: ${archivedNotification.providerId}`,
+      );
+
+      return archivedNotification;
+    });
+  }
+
+  async moveCompletedNotificationsToArchive(): Promise<void> {
+    const archiveLimit = this.configService.get<number>('ARCHIVE_LIMIT', 1000);
+
+    try {
+      // Step 1: Retrieve the notifications to archive
+      this.logger.log(`Retrieve the top ${archiveLimit} notifications to archive`);
+      const notificationsToArchive =
+        await this.notificationsService.findNotificationsToArchive(archiveLimit);
```

Comment:
This need to be done.  @kshitij-k-osmosys

Reply:
We will be archiving only that entry that has been retrieved in the array. Since success/failed is final status it will not change. This also avoids stopping of queryrunner when no notificatiosn to archive.

But if needed working on a fix as suggested

Reply:
fixed 99f26885823633d7246715e18738663e4608eb0c